test/rtree.c

```diff
@@ -0,0 +1,1063 @@
+/* * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * *
+ * Copyright by The HDF Group.                                               *
+ * All rights reserved.                                                      *
+ *                                                                           *
+ * This file is part of HDF5.  The full HDF5 copyright notice, including    *
+ * terms governing use, modification, and redistribution, is contained in    *
+ * the COPYING file, which can be found at the root of the source code      *
+ * distribution tree, or in https://www.hdfgroup.org/licenses.              *
+ * If you do not have access to either file, you may request a copy from    *
+ * help@hdfgroup.org.                                                        *
+ * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * */
+
+/*
+ * Programmer:  Matthew Larson
+ *              September 9, 2025
+ *
+ * Purpose:     Test the R-tree spatial indexing implementation.
+ */
+
+#include "h5test.h"
+
+/*
+ * This file needs to access private datatypes from the H5RT package.
+ */
+#define H5RT_FRIEND /*suppress error about including H5RTpkg */
+#define H5RT_TESTING
```

Comment:
Minor note: developer builds show that this macro isn't used. It also seemed like `H5CXprivate.h` and `H5VLprivate.h` below aren't used either.